### DISCUSSION TITLE
Deploy on Azure via Terraform

### DISCUSSION THREAD

**Discussion by LarryBambelle** [danny-avila/LibreChat#1671] [Troubleshooting] [answered]
### What happened?

I recently deployed LibreChat on Azure following the instructions in the wiki. The deployment was successful, and I was able to access the website, create an account, and log in. However, I'm encountering an issue shortly after logging in.

Issue Description:

After successfully logging in, the UI is visible for a few seconds before an unexpected application error occurs.
Error Message: "Unexpected Application Error! Cannot read properties of undefined (reading 'sort')"
It appears to be a TypeError related to the 'sort' property.
Error Details:

The error seems to originate from a script file (index-eG_k48uu.js) and is possibly related to some vendor scripts.
Specific script references in the error log have been omitted for privacy.
I'm looking for guidance or suggestions on how to resolve this issue. Any help would be greatly appreciated.

### Steps to Reproduce

1. Deploy on Azure according Wiki
2. Create and account
3. Log in
I can send URL if needed.

### Relevant log output

```shell
Unexpected Application Error!
Cannot read properties of undefined (reading 'sort')
TypeError: Cannot read properties of undefined (reading 'sort')
    at https://********.azurewebsites.net/assets/index-eG_k48uu.js:37:88412
    at gg (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:24283)
    at $c (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:42409)
    at nz (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:41231)
    at zl (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:40278)
    at a7 (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:36886)
    at wl (https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:38:3279)
    at https://********.azurewebsites.net/assets/vendor-XGrjyrDJ.js:40:34263
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It's hard to debug this without being able to see the mapped trace output. To do so, you would need to be able to run `npm run frontend:dev`, which creates a proxy to a dev build, which includes hot reloads to frontend code changes, as well as completely mapped output. in the same environment the LibreChat server is running.

You can also edit this line and then rebuild the app/image: https://github.com/danny-avila/LibreChat/blob/f7f7f929a03021a99ad344d6a3231c4136cd79ff/client/vite.config.ts#L30

```js
  build: {
    sourcemap: true, // changed this to `true` replacing `process.env.NODE_ENV === 'development'`
```

After doing so, we can get a more meaningful error to see where/why the error occurs.

The terraform instructions are maintained by @thunderbug1 in a separate repo: https://github.com/thunderbug1/LibreChatAzureDeployment. 

I can't comment on the setup as I haven't personally deployed it.

Note that the `.sort` native Array method is used sparingly, and almost exclusively when dealing with MongoDB documents. If you are using CosmosDB, that could be suspect of the cause.

**Comment by LarryBambelle** [comment]
Thanks for the quick answer.
I will look further from this informations

**Comment by LarryBambelle** [comment, marked as answer]
Thanks for everything.
I don't why, I just had to delete Users table and recreate it and it worked... 

If possible, do you have any idea why would this be the case  ?
If not, this case can be closed. Thanks again.

**Reply by danny-avila** [reply]
Glad it's working! I have a hunch why this might be, are you using CosmosDB?
